### PR TITLE
UI/Add activity location on activity card and improve  empty Activities Screen.

### LINKED ISSUE #310: display activity location on Activity Card

### PR DESCRIPTION
**Pull request by 1andreea26**
## Summary

This short PR introduces 2 UI improvements:
- The Activity Card now displays the location name of the activity (if the activity has a set location).
- The Activities Screen now has a prompt for when the screen is empty, not displaying the `Drafts` and `Final` titles anymore/the price.
## Changes

- **Screens/UI:** Activities Screen now displays empty prompt in case of no activities, activity card now displays location of activities.
- **Bug Fixes/Improvements:** Addresses #310.

## Checklist

- [X] This PR resolves #310.
- [X] Tests have been added for new functionality.
- [X] Screenshots or UI changes have been attached (if applicable).
- [X] Documentation has been updated (if applicable).

##  Testing

- **Manual Testing:**
    - [X] Tested on emulator / physical device.
- **Unit Tests**
    - [X] Added units tests for this

##  Related Issues/Tickets

Closes #310.

## Screenshots (if applicable)

Screenshot 2024-12-16 at 17 59 18
Screenshot 2024-12-16 at 18 00 17

## 💡 Additional Context

The tests class has been updated, and some tests where the `Drafts` and `Final `titles were checked (even when there are no activities) have been updated/removed (now, there is a test which checks if the empty prompt is displayed).

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -91,6 +91,12 @@
 
 
  <!-- Add Activity -->
+    <!-- Activities -->
+    <string name="no_activities_scheduled">"No activities have been scheduled yet."</string>
+    <!-- Activity Item -->
+    <string name="location_text">Location</string>
+
+    <!-- Add Activity -->
```
Comment (ioanajianu) [inline app/src/main/res/values/strings.xml:99]: nit: I'm a bit confused about the comments (Add Activity appearing twice, for example). In case we use some strings in both the Add Activity and Activities screen, can we leave a single comment `<!-- Activities -->` to avoid confusion? Let me know what you think

File: app/src/main/java/com/android/voyageur/ui/trip/activities/Activities.kt [hunk 2]
```diff
@@ -176,86 +177,96 @@ fun ActivitiesScreen(
...
-                    isEditable,
-                    onClickButton = {
-                      activityToDelete = activity
-                      showDialog = true
-                    },
-                    buttonType,
-                    navigationActions,
-                    tripsViewModel)
-                Spacer(modifier = Modifier.height(10.dp))
+              Text(
```
Comment (ioanajianu) [inline app/src/main/java/com/android/voyageur/ui/trip/activities/Activities.kt:196]: I was thinking, what if we show "Drafts" only if we currently have draft activities? Same for "Final". That way we avoid having one final activity and the screen showing 
Drafts
Final
Activity

**Review by ioanajianu — COMMENTED** [review]
Thanks Andreea for implementing these fixes! The code looks good, and the line coverage is 100%! Great work! I just left a few comments, let me know what you think.

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (1andreea26) [reply app/src/main/res/values/strings.xml:99]: Thanks for pointing this out, this was a merge error. I'll leave only the `<!-- Activities --> ` comment.

File: app/src/main/java/com/android/voyageur/ui/trip/activities/Activities.kt (hunk 2 above)
Comment (1andreea26) [reply app/src/main/java/com/android/voyageur/ui/trip/activities/Activities.kt]: Fixed.

File: app/src/main/java/com/android/voyageur/ui/formFields/UserDropdown.kt [hunk 3]
```diff
@@ -99,9 +99,10 @@ fun UserDropdown(
             DropdownMenuItem(
                 text = { Text("${userPair.first.name} ") },
                 onClick = { onUpdate(userPair, index) },
-                leadingIcon = {
-                  Checkbox(
-                      checked = userPair.second, onCheckedChange = { onUpdate(userPair, index) })
+                trailingIcon = {
+                  Button(onClick = {}) { Text("Send trip request") }
+                  // Checkbox(
+                  ///    checked = userPair.second, onCheckedChange = { onUpdate(userPair, index) })
```
Comment (ioanajianu) [inline app/src/main/java/com/android/voyageur/ui/formFields/UserDropdown.kt:105]: why is this change needed here?

Comment (1andreea26) [reply app/src/main/java/com/android/voyageur/ui/formFields/UserDropdown.kt:105]: It's not, thanks for noticing! This was for the other feature I am working on, I think I forgot to change branches... I'll revert this..

**Review by ioanajianu — APPROVED** [review]
Thanks for implementing my suggestions! I tested your changes, and everything works well. Thanks for this fix! Approved!
